lib.rs

```diff
@@ -403,11 +403,13 @@ impl<A: Array> SmallVec<A> {
     /// Construct an empty vector
     #[inline]
     pub fn new() -> SmallVec<A> {
-        unsafe {
-            SmallVec {
-                capacity: 0,
-                data: SmallVecData::from_inline(mem::uninitialized()),
-            }
+        debug_assert!(
```

Comment:
This is an attempt to catch any obviously invalid custom implementations of `Array`. I'd rather do this statically, but can't find a way to, since the compiler won't let me use the sizes of the generic parameters statically :(

Reply:
You can go and use an `assert!` there: the condition will be trivially true or trivially false after monomorphisation, so it will get elided if the test passes

Reply:
Done.